docs/source/training.rst

```diff
@@ -10,290 +10,362 @@
+:class:`~transformers.Trainer` to compute any metrics. Let's have a look on how to do that now!
 
-Then all we have to do is call ``scheduler.step()`` after ``optimizer.step()``.
+To have the :class:`~transformers.Trainer` compute and report metrics, we need to give it a :obj:`compute_metrics`
+function that takes predictions and labels (grouped in a namedtuple called :class:`~transformers.EvalPrediction`) and
+return a dictionary with string items (the metric names) and float values (the metric values).
+
+The 🤗 Datasets library provides an easy way to get the common metrics used in NLP with the :obj:`load_metric` function.
+here we simply use accuracy. Then we define the :obj:`compute_metrics` function that just convert logits to predictions
+(remember that all 🤗 Transformers models return the logits) and feed them to :obj:`compute` method of this metric.
 
 .. code-block:: python
 
-    loss.backward()
-    optimizer.step()
-    scheduler.step()
+    import numpy as np
+    from datasets import load_metric
+
+    metric = load_metric("accuracy")
 
-We highly recommend using :func:`~transformers.Trainer`, discussed below, which conveniently handles the moving parts
-of training 🤗 Transformers models with features like mixed precision and easy tensorboard logging.
+    def compute_metrics(eval_pred):
+        logits, labels = eval_pred
+        predictions = np.argmax(logits, axis=-1)
+        return metric.compute(predictions=predictions, references=labels)
 
+To check if this works on practice, let's create a new :class:`~transformers.Trainer` with our fine-tuned model:
 
-Freezing the encoder
------------------------------------------------------------------------------------------------------------------------
+.. code-block:: python
+
+    trainer = Trainer(
+        model=model,
+        args=training_args,
+        train_dataset=small_train_dataset,
+        eval_dataset=small_eval_dataset,
+        compute_metrics=compute_metrics,
+    )
+    trainer.evaluate()
+
+which showed an accuracy of 87.5% in our case.
 
-In some cases, you might be interested in keeping the weights of the pre-trained encoder frozen and optimizing only the
-weights of the head layers. To do so, simply set the ``requires_grad`` attribute to ``False`` on the encoder
-parameters, which can be accessed with the ``base_model`` submodule on any task-specific model in the library:
+If you want to fine-tune your model and regularly report the evaluation metrics (for instance at the end of each
+epoch), here is how you should define your training arguments:
 
 .. code-block:: python
 
-    for param in model.base_model.parameters():
-        param.requires_grad = False
+    from transformers import TrainingArguments
 
+    training_args = TrainingArguments("test_trainer", evaluation_strategy="epoch")
 
-.. _tensorflow:
+See the documentation of :class:`~transformers.TrainingArguments` for all the options you can tweak with it.
```

Comment:
```suggestion
See the documentation of :class:`~transformers.TrainingArguments` for more options.
```

Reply:
Do you think that it is worth mentioning the `DataCollatorForTokenClassification`?  I thought that was extremely useful?  Ore even perhaps it might be worth linking to your example notebooks somewhere in this document in order to learn more?